src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventDeadLetterQueueListener.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.hmcts.reform.sscs.config;
+
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessage;
+import com.azure.messaging.servicebus.ServiceBusReceiverClient;
+import com.azure.messaging.servicebus.models.SubQueue;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import uk.gov.hmcts.reform.sscs.model.hearings.HearingRequest;
+import uk.gov.hmcts.reform.sscs.service.AppInsightsService;
+
+@Slf4j
+@Configuration
+@RequiredArgsConstructor
+@ConditionalOnProperty("flags.tribunals-to-hearings-api.enabled")
+public class TribunalsHearingsEventDeadLetterQueueListener {
+
+    private final TribunalsHearingEventQueueProperties tribunalsHearingEventQueueProperties;
+    private final AppInsightsService appInsightsService;
+    private boolean keepRun = true;
+
+    @Bean
+    public void processDeadLetterQueue() {
+        try (ServiceBusReceiverClient sessionReceiver = tribunalsHearingsEventProcessorClientDeadLetter()) {
+            while (keepRun) {
+                consumeMessage(sessionReceiver);
+            }
+        } catch (Exception ex) {
+            log.error("Error Processing Queue: " + ex);
+        }
```

Comment:
Catching "Exception" is a bold thing to do, especially when only logging it.
Consider what else you may need to do. Or even consider the possible exceptions that could be thrown and potentially only catch checked exceptions to allow runtime ones to flow up and be handled higher up.
Where you fully intend on catching "Exception" I'd suggest you make this clear so other dev's better understand

Reply:
Resolved